commands/init.go

```diff
@@ -299,81 +257,34 @@ func (a *initFnCmd) init(c *cli.Context) error {
 				return err
 			}
 		}
+	}
 
-		if err := common.EncodeFuncFileV20180708YAML("func.yaml", a.ff); err != nil {
-			return err
-		}
+	if err := common.EncodeFuncFileV20180708YAML("func.yaml", a.ff); err != nil {
+		return err
 	}
 
 	fmt.Println("func.yaml created.")
 	return nil
 }
 
-func runInitImage(initImage string, a *initFnCmd) error {
-	fmt.Println("Building from init-image: " + initImage)
-
-	// Run the initImage
-	var c1ErrB bytes.Buffer
-	tarR, tarW := io.Pipe()
-
-	c1 := exec.Command("docker", "run", "--rm", "-e", "FN_FUNCTION_NAME="+a.ff.Name, initImage)
-	c1.Stderr = &c1ErrB
-	c1.Stdout = tarW
-
-	c1Err := c1.Start()
-	if c1Err != nil {
-		fmt.Println(c1ErrB.String())
-		return errors.New("Error running init-image")
+func (a *initFnCmd) doInitImage(err error, initImage string, c *cli.Context) error {
```

Comment:
Why do you take an error as a parameter then immediately overwrite it here? Since this is unexported and we don't need to match any specific function signature can we just remove the parameter?

Reply:
Good catch. I blame Goland refactoring :)